python/lsst/ip/diffim/dipoleMeasurement.py

```diff
@@ -23,34 +23,78 @@
 import lsst.afw.geom as afwGeom
 import lsst.afw.image as afwImage
 import lsst.afw.detection as afwDetect
-import lsst.pipe.base as pipeBase
 import lsst.pex.logging as pexLog
 import lsst.pex.config as pexConfig
 import lsst.meas.deblender.baseline as deblendBaseline
-from lsst.meas.base import SingleFrameMeasurementTask, SingleFrameMeasurementConfig
+from lsst.meas.base.pluginRegistry import register
+from lsst.meas.base import SingleFrameMeasurementTask, SingleFrameMeasurementConfig, \
+    SingleFramePluginConfig, SingleFramePlugin
 import lsst.afw.display.ds9 as ds9
 
 __all__ = ("DipoleMeasurementConfig", "DipoleMeasurementTask", "DipoleAnalysis", "DipoleDeblender",
-           "SourceFlagChecker")
+           "SourceFlagChecker", "ClassificationDipoleConfig", "ClassificationDipolePlugin")
 
 
-class DipoleClassificationConfig(pexConfig.Config):
-    """!Classification of detected diaSources as dipole or not"""
+class ClassificationDipoleConfig(SingleFramePluginConfig):
+    """Configuration for classification of detected diaSources as dipole or not"""
     minSn = pexConfig.Field(
         doc="Minimum quadrature sum of positive+negative lobe S/N to be considered a dipole",
         dtype=float, default=np.sqrt(2) * 5.0,
         )
     maxFluxRatio = pexConfig.Field(
-        doc = "Maximum flux ratio in either lobe to be considered a dipole",
+        doc="Maximum flux ratio in either lobe to be considered a dipole",
         dtype = float, default = 0.65
         )
 
+
+@register("ip_diffim_ClassificationDipole")
+class ClassificationDipolePlugin(SingleFramePlugin):
+    """
+    A binary measure of the whether a diasource is a dipole.
+    """
+
+    ConfigClass = ClassificationDipoleConfig
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.CLASSIFY_ORDER
+
+    def __init__(self, config, name, schema, metadata):
+        SingleFramePlugin.__init__(self, config, name, schema, metadata)
+        self.dipoleAnalysis = DipoleAnalysis()
+        self.keyProbability = schema.addField(name + "_value", type="D",
```

Comment:
Interesting... I gave some serious thought to the relationship between classification and probability  -- more than is necessary than deciding a name for an instance variable. It is a number between 0.0 and 1.0 inclusive which makes it a probability in the technical sense.

I'm not a fan of the name classification because that's the name of the plugin. This classification currently has two components: an overall probability and a flag. I'd prefer {{self.keyValue}}

Also, Becker's original implementation (just a hook for a better implementation) sets a float with the intention of leaving the door open for the general implementation returning a float between 0 and 1. As a user, I'd like the field to be a float so that I can chose my own cut off for dipoles depending on the application.